### PR TITLE
Added libraries vue to Credits

### PR DESCRIPTION
**Pull request by areksolek**
Added libraries vue to Credits

### CODE REVIEW

File: app/Installer/Composer.php [hunk 1]
```diff
@@ -174,6 +174,33 @@ public static function install(\Composer\Script\Event $event)
...
+	 * Parse credits vue.
+	 */
+	public static function parseCreditsVue()
+	{
+		$rootDir = realpath(__DIR__ . '/../../') . \DIRECTORY_SEPARATOR . 'public_html' . \DIRECTORY_SEPARATOR;
+		$dirLibraries = $rootDir . 'src' . \DIRECTORY_SEPARATOR . 'node_modules' . \DIRECTORY_SEPARATOR;
+		$dataEncode = [];
+		foreach (new \DirectoryIterator($dirLibraries) as $level1) {
+			$fileName = $level1->getFilename();
+			if ('.' === $fileName || '..' === $fileName) {
```
Comment (rskrzypczak) [inline app/Installer/Composer.php]: ->isDot()

File: app/Installer/Credits.php [hunk 2]
```diff
@@ -124,6 +124,37 @@ public static function getPublicLibraries()
...
+		$libraries = [];
+		$dir = ROOT_DIRECTORY . \DIRECTORY_SEPARATOR . 'public_html' . \DIRECTORY_SEPARATOR . 'src' . \DIRECTORY_SEPARATOR;
+		if (file_exists($dir . 'libraries.json')) {
+			$yarnFile = \App\Json::decode(file_get_contents($dir . 'libraries.json'));
+			foreach ($yarnFile as $nameWithVersion) {
+				foreach ($nameWithVersion as $nameLibraries => $verisonWithLicence) {
+					$isPrefix = 0 === strpos($nameLibraries, '@');
+					$name = $isPrefix ? '@' : '';
+					$tempName = explode('@', $isPrefix ? ltrim($nameLibraries, '@') : $nameLibraries);
+					$name .= array_shift($tempName);
```
Comment (rskrzypczak) [inline app/Installer/Credits.php]:
```suggestion
					$name .= current($tempName);
```

File: app/Installer/Credits.php [hunk 3]
```diff
@@ -124,6 +124,37 @@ public static function getPublicLibraries()
...
 	}
 
+	/**
+	 * Function gets vue libraries name.
+	 *
+	 * @throws \App\Exceptions\AppException
+	 *
+	 * @return array
+	 */
+	public static function getVueLibs()
```
Comment (rskrzypczak) [inline app/Installer/Credits.php]: ?

File: app/Installer/Composer.php [hunk 4]
```diff
@@ -174,6 +174,33 @@ public static function install(\Composer\Script\Event $event)
...
+			try {
+				if ($getValues = Credits::parseLibraryVueValues($fileName, $dirLibraries)) {
+					$value[$getValues['name']] = ['version' => $getValues['version'], 'license' => $getValues['license'], 'homepage' => $getValues['homepage']];
+					$dataEncode[] = $value;
+				}
+			} catch (\Exception $ex) {
+				\App\Log::warning($ex->getMessage());
+			}
+		}
+		file_put_contents($rootDir . \DIRECTORY_SEPARATOR . 'src' . \DIRECTORY_SEPARATOR . 'libraries.json', \App\Json::encode($dataEncode));
```
Comment (mariuszkrzaczkowski) [inline app/Installer/Composer.php]: /cache/libraries.json

File: app/Installer/Credits.php [hunk 5]
```diff
@@ -124,6 +124,37 @@ public static function getPublicLibraries()
...
+	{
+		$libraries = [];
+		$dir = ROOT_DIRECTORY . \DIRECTORY_SEPARATOR . 'public_html' . \DIRECTORY_SEPARATOR . 'src' . \DIRECTORY_SEPARATOR;
+		if (file_exists($dir . 'libraries.json')) {
+			$yarnFile = \App\Json::decode(file_get_contents($dir . 'libraries.json'));
```
Comment (mariuszkrzaczkowski) [inline app/Installer/Credits.php]:
```suggestion
			$yarnFile = \App\Json::read($dir . 'libraries.json');
```

File: app/Installer/Credits.php [hunk 6]
```diff
@@ -124,6 +124,37 @@ public static function getPublicLibraries()
...
+			foreach ($yarnFile as $nameWithVersion) {
```
Comment (mariuszkrzaczkowski) [inline app/Installer/Credits.php]:
```suggestion
			foreach (\App\Json::read($dir . 'libraries.json') as $nameWithVersion) {
```

File: app/Installer/Credits.php [hunk 7]
```diff
@@ -164,6 +195,42 @@ public static function getLibraryValues($name, $dir)
...
+	 *
+	 * @return array
+	 */
+	public static function parseLibraryVueValues($name, $dir)
+	{
+		$library = ['name' => $name, 'version' => '', 'license' => '', 'homepage' => ''];
+		foreach (self::$jsonFiles as $file) {
+			$packageFile = $dir . $name . \DIRECTORY_SEPARATOR . $file;
+			if (file_exists($packageFile)) {
+				$packageFileContent = \App\Json::decode(file_get_contents($packageFile), true);
```
Comment (mariuszkrzaczkowski) [inline app/Installer/Credits.php]:
```suggestion
				$packageFileContent = \App\Json::read($packageFile);
```

File: app/Installer/Composer.php [hunk 8]
```diff
@@ -174,6 +174,35 @@ public static function install(\Composer\Script\Event $event)
...
+					if ($getValues = Credits::parseLibraryVueValues($fileName, $dirLibraries)) {
+						$value[$getValues['name']] = ['version' => $getValues['version'], 'license' => $getValues['license'], 'homepage' => $getValues['homepage']];
+						$dataEncode[] = $value;
+					}
+				} catch (\Exception $ex) {
+					\App\Log::warning($ex->getMessage());
+				}
+			}
+		}
+		file_put_contents($rootDir . 'cache' . \DIRECTORY_SEPARATOR . 'libraries.json', \App\Json::encode($dataEncode));
```
Comment (mariuszkrzaczkowski) [inline app/Installer/Composer.php]:
```suggestion
		\App\Json::encode($rootDir . 'cache' . \DIRECTORY_SEPARATOR . 'libraries.json', $dataEncode);
```

**Pull request merged**